controllers/productController.js

```diff
@@ -129,7 +129,28 @@ export const productPhotoController = async (req, res) => {
 //delete controller
 export const deleteProductController = async (req, res) => {
   try {
-    await productModel.findByIdAndDelete(req.params.pid).select("-photo");
+    const { pid } = req.params;
+
+    // check if pid is null
+    if (!pid) {
```

Comment:
I think this is ok but if you still want to add more checks, you can check whether the pid conforms to certain format for the 12 bytes

Reply:
Okay noted  on that, have refined these in the new commit

Reply:
LGTM!